raft/raft.go

```diff
@@ -33,12 +33,13 @@ const None uint64 = 0
 const noLimit = math.MaxUint64
 
 // Possible values for StateType.
+type StateType string
+
 const (
-	StateFollower StateType = iota
-	StateCandidate
-	StateLeader
-	StatePreCandidate
-	numStates
+	StateFollower     StateType = "StateFollower"
```

Comment:
does this mean every state comparison will now be a string comparison? if so, seems like a lot of overhead for little benefit